c++/src/kj/compat/http.c++

```diff
@@ -1956,8 +1956,13 @@ public:
       if (length > 0) {
         // We haven't reached the end of the entity body yet.
         if (amount == 0) {
-          kj::throwRecoverableException(KJ_EXCEPTION(DISCONNECTED,
-              "premature EOF in HTTP entity body; did not reach Content-Length"));
+          size_t expectedLength = length + alreadyRead;
```

Comment:
My only challenge with calculating it is that if any bugs sneak in calculating these it might not actually match the value passed in. Not a big deal tho given that such a bug would likely be noticeable elsewhere.